src/Advection/adapt_advection_order.jl

```diff
@@ -39,9 +41,9 @@ function adapt_advection_order(advection, grid::AbstractGrid)
 end
 
 function adapt_advection_order(advection::FluxFormAdvection, grid::AbstractGrid)
-    advection_x = adapt_advection_order_x(advection.x, grid, grid.Nx)
-    advection_y = adapt_advection_order_y(advection.y, grid, grid.Ny)
-    advection_z = adapt_advection_order_z(advection.z, grid, grid.Nz)
+    advection_x = adapt_advection_order_x(advection.x, topology(grid, 1), size(grid, 1), grid)
+    advection_y = adapt_advection_order_y(advection.y, topology(grid, 2), size(grid, 2), grid)
+    advection_z = adapt_advection_order_z(advection.z, topology(grid, 3), size(grid, 3), grid)
```

Comment:
And now you don't need _x, _y, _z right?

Reply:
We need it because of WENO schemes that might have stretched directions in x, y or z so we have to check if we need to reconstruct the scheme using grid information or not.

Reply:
probably we can just check that any direction is stretched. Then we use the grid.

Reply:
I see, it's not a topology thing